### PR TITLE
HRINT-1715 - licensing adjustments for TimeSeries

### PR DESCRIPTION
**Pull request by grisha-kotler**

### CODE REVIEW

**Review by ayende — APPROVED** [review]

File: src/Raven.Client/Exceptions/Commercial/LimitType.cs [hunk 1]
```diff
@@ -61,6 +55,9 @@ public enum LimitType
         [Description("Pull Replication As Sink")]
         PullReplicationAsSink,
 
+        [Description("Time Series Aggregation and Retention")]
+        TimeSeriesAggregationAndRetention,
```
Comment (ppekrol) [inline src/Raven.Client/Exceptions/Commercial/LimitType.cs:59]: TimeSeriesRollupsAndRetention

File: src/Raven.Server/Commercial/LicenseManager.cs [hunk 2]
```diff
@@ -977,6 +982,12 @@ private void ThrowIfCannotActivateLicense(LicenseStatus newLicenseStatus)
                 throw GenerateLicenseLimit(LimitType.PullReplicationAsSink, message);
             }
 
+            if (timeSeriesAggregationAndRetentionCount > 0 && newLicenseStatus.HasTimeSeriesAggregationAndRetention == false)
+            {
+                var message = GenerateDetails(timeSeriesAggregationAndRetentionCount, "time series aggregation and retention");
```
Comment (ppekrol) [inline src/Raven.Server/Commercial/LicenseManager.cs:987]: rollups

File: src/Raven.Server/Commercial/LicenseManager.cs [hunk 3]
```diff
@@ -1192,6 +1211,21 @@ public void AssertCanAddPullReplicationAsSink()
...
+            if (IsValid(out var licenseLimit) == false)
+                throw licenseLimit;
+
+            if (LicenseStatus.HasTimeSeriesAggregationAndRetention)
+                return;
+
+            if (HasTimeSeriesAggregationAndRetention(configuration) == false)
+                return;
+
+            var details = $"Your current license ({LicenseStatus.Type}) does not allow adding time series aggregation and retention";
```
Comment (ppekrol) [inline src/Raven.Server/Commercial/LicenseManager.cs:1225]: rollups

File: src/Raven.Server/Commercial/LicenseValidator.cs [hunk 4]
```diff
@@ -17,7 +17,7 @@ public static class LicenseValidator
             "distributedCluster", "maxClusterSize", "snapshotBackup", "cloudBackup", "dynamicNodesDistribution",
             "externalReplication", "delayedExternalReplication", "ravenEtl", "sqlEtl", "highlyAvailableTasks",
             "pullReplicationAsHub", "pullReplicationAsSink", "encryptedBackup", "letsEncryptAutoRenewal", "cloud",
-            "documentsCompression"
+            "documentsCompression", "timeSeriesAggregationAndRetention"
```
Comment (ppekrol) [inline src/Raven.Server/Commercial/LicenseValidator.cs:20]: rollups

**Review by ppekrol — CHANGES_REQUESTED** [review]

**Comment by ppekrol** [comment]
@karmeli87 will apply the fixes. Merging.

**Pull request merged**